src/components/combobox/combobox.tsx

```diff
@@ -382,6 +382,12 @@ export class Combobox implements LabelableComponent, FormComponent, InteractiveC
 
   keydownHandler = (event: KeyboardEvent): void => {
     const { key } = event;
+    const bounding = this.el.getBoundingClientRect();
+    const inViewport =
```

Comment:
Can you refactor the viewport-checking code to only run when needed (i.e., to run when the pressed key is related to item navigation)?